File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ind = None
        for i in range(len(nums)):
            if nums[i]==0:
                ind = i
                break
        if ind is None:
            return nums
        
        a = ind
        j = ind+1
        while j<len(nums):
            if nums[j] == 0:
                j+=1
            else:
                nums[a],nums[j] = nums[j],nums[a]

                
                while a<=j:
                    if nums[a] == 0:
                        break
                    a+=1
                j+=1
        return nums

File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {

        int n = nums.size();


        int k =0;

        for(int i = k;i<n-1 ;i++){

            if(i>=n-1) break;

            if (nums[i] == 0){

        for (int j = i+1;;j++){

            if (nums[j]==0) ;

            else{
                 swap(nums[i], nums[j]);
                 break;     
                 }

            if (j>=n-1) break;     

        }


        }
        }

        
        
        
        
        
    }
};